### PR TITLE
feat(gen5): Wave 6A -- status/utility move effects and master dispatcher

### PR DESCRIPTION
**Pull request by uehlbran**
## Summary
- Add `Gen5MoveEffectsStatus.ts` with 8 move handlers: Heal Pulse, Aromatherapy, Heal Bell, Soak, Incinerate, Bestow, Entrainment, Round
- Replace `Gen5MoveEffects.ts` stub with full master dispatch chain (field -> behavior -> combat -> status -> BaseRuleset fallback)
- Wire all 4 sub-modules into `Gen5Ruleset.executeMoveEffect()` (previously only had field + combat)
- Update `index.ts` to export all sub-module public symbols
- Add 38 tests in `move-effects-status.test.ts` covering all moves, edge cases, and dispatch integration

Key Gen 5 vs Gen 6+ differences implemented:
- **Heal Pulse**: always 50% heal via `Math.ceil` (no Mega Launcher 75% boost)
- **Aromatherapy/Heal Bell**: bypass Soundproof (cure all allies unconditionally)
- **Soak**: succeeds on Water-type targets (Gen 6+ fails on pure Water)
- **Incinerate**: destroys Berries only (Gen 6+ also destroys Gems)

Closes: N/A

## Test plan
- [x] All 38 new tests pass in `move-effects-status.test.ts`
- [x] All 820 existing gen5 tests pass (no regressions)
- [x] Full monorepo test suite passes
- [x] TypeScript typecheck passes
- [x] Biome lint/format passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Implemented Gen 5 status/utility move effects including Heal Pulse, Aromatherapy, Heal Bell, Soak, Incinerate, Bestow, Entrainment, and Round with proper battle mechanics.
  * Added support for defender healing, team-wide status curing, and ability changes from move effects.
  * Enhanced Round move power calculation for Gen 5 battles.

* **Tests**
  * Added comprehensive test coverage for new move effects and battle engine mechanics.

### CODE REVIEW

File: packages/gen5/tests/move-effects-status.test.ts [hunk 1]
```diff
@@ -0,0 +1,787 @@
...
+    expect(isBerry(null)).toBe(false);
+  });
+
+  it("given undefined, when checked, then returns false", () => {
+    expect(isBerry(undefined)).toBe(false);
+  });
+
+  it("given empty string, when checked, then returns false", () => {
+    expect(isBerry("")).toBe(false);
+  });
```
[the remark this answers is not in the document]
Comment (uehlbran) [reply packages/gen5/tests/move-effects-status.test.ts]: Fixed in 0c8d58e: Added derivation comments.

File: packages/gen5/src/Gen5MoveEffectsStatus.ts [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+      messages: ["But it failed!"],
+    });
+  }
+
+  // Success: signal ability change via message. The engine handles the actual
+  // ability mutation. We signal it via a message that the engine can parse.
+  return makeResult({
+    messages: [`${ctx.defender.pokemon.nickname ?? "The target"} acquired ${sourceAbility}!`],
+  });
+}
```
[the remark this answers is not in the document]
Comment (uehlbran) [reply packages/gen5/src/Gen5MoveEffectsStatus.ts]: Fixed in 0c8d58e: Added `abilityChange` optional field to `MoveEffectResult` with shape `{ target: 'attacker' | 'defender', ability: string }`. Engine mutates `target.ability` in `processEffectResult` and emits a message event. Updated `handleEntrainment` to return `abilityChange: { target: 'defender', ability: sourceAbility }` on success. Engine-level test verifies the defender's ability is changed. Success test now asserts `abilityChange` field.

Source: Showdown data/moves.ts `entrainment: target.setAbility(source.ability)`, Bulbapedia "Entrainment changes the target's Ability to match the user's".
